File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/interfaces/ErrorType.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.devsupport.interfaces;

import com.facebook.infer.annotation.Nullsafe;

@Nullsafe(Nullsafe.Mode.LOCAL)
public enum ErrorType {
  JS("JS"),
  NATIVE("Native");

  private final String name;

  ErrorType(String name) {
    this.name = name;
  }

  public String getName() {
    return name;
  }
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/interfaces/ErrorType.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.devsupport.interfaces

public enum class ErrorType(public val displayName: String) {
  JS("JS"),
  NATIVE("Native");

  override fun toString(): String = displayName
}
